### PR TITLE
adding other alternate AWS builds to circleCI

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

 /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

**What this PR does / why we need it**:

Uploads other Falco images `no-driver` and `driver-loader` to AWS public gallery.

**Does this PR introduce a user-facing change?**:

No

```release-note
new: falco-no-driver container images on AWS ECR gallery
new: falco-driver-loader container images on AWS ECR gallery
```

### CODE REVIEW

**Review by leodido — COMMENTED** [review]
Did an initial review and suggested a fix that I'll apply.

It's late now, I'll continue tomorrow...

/hold

**Comment by jonahjon** [comment]
Change looks good!

File: .circleci/config.yml [hunk 1]
```diff
@@ -546,6 +568,17 @@ jobs:
           at: /
       - checkout
       - setup_remote_docker
+      - run:
+          name: Build and publish no-driver to AWS
+          command: |
+            apk update
+            apk add --update groff less py-pip
+            pip install awscli
+            docker build --build-arg VERSION_BUCKET=deb --build-arg FALCO_VERSION=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" docker/no-driver
```
Comment (leodido) [inline .circleci/config.yml:577]:
```suggestion
            docker build --build-arg VERSION_BUCKET=bin --build-arg FALCO_VERSION=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" docker/no-driver
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -546,6 +568,17 @@ jobs:
...
+            docker tag "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" public.ecr.aws/falcosecurity/falco-no-driver:latest
+            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
+            docker push "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}"
+            docker push "public.ecr.aws/falcosecurity/falco-no-driver:latest"
```
Comment (leodido) [inline .circleci/config.yml:581]: Shouldn't we tag this image as `falco:${CIRCLE_TAG}-slim` and `falco:latest-slim` too?

**Review by leodido — COMMENTED** [review]
Left another comment Jonah

File: .circleci/config.yml (hunk 2 above)
Comment (leodido) [reply .circleci/config.yml:581]: And push'em too?

File: .circleci/config.yml [hunk 3]
```diff
@@ -546,6 +568,17 @@ jobs:
...
       - checkout
       - setup_remote_docker
+      - run:
+          name: Build and publish no-driver to AWS
+          command: |
+            apk update
+            apk add --update groff less py-pip
+            pip install awscli
+            docker build --build-arg VERSION_BUCKET=bin --build-arg FALCO_VERSION=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" docker/no-driver
+            docker tag "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" public.ecr.aws/falcosecurity/falco-no-driver:latest
```
Comment (leodido) [inline .circleci/config.yml:578]:
```suggestion
            docker tag "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" public.ecr.aws/falcosecurity/falco-no-driver:latest
            docker tag "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}-slim"
            docker tag "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}" "public.ecr.aws/falcosecurity/falco:latest-slim"
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -546,6 +568,17 @@ jobs:
...
+            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
+            docker push "public.ecr.aws/falcosecurity/falco-no-driver:${CIRCLE_TAG}"
+            docker push "public.ecr.aws/falcosecurity/falco-no-driver:latest"
```
Comment (leodido) [inline .circleci/config.yml:581]:
```suggestion
            docker push "public.ecr.aws/falcosecurity/falco-no-driver:latest"
            docker push "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}-slim"
            docker push "public.ecr.aws/falcosecurity/falco:latest-slim"
```

File: .circleci/config.yml (hunk 2 above)
Comment (leodido) [reply .circleci/config.yml:581]: Left suggestions for what mentioned here yesterday

File: .circleci/config.yml [hunk 5]
```diff
@@ -557,6 +594,17 @@ jobs:
             aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
             docker push "public.ecr.aws/falcosecurity/falco:${CIRCLE_TAG}"
             docker push "public.ecr.aws/falcosecurity/falco:latest"
+      - run:
+          name: Build and publish driver-loader to AWS
```
Comment (leodido) [inline .circleci/config.yml:598]:
```suggestion
          name: Build and publish falco-driver-loader to AWS
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -557,6 +594,17 @@ jobs:
...
+          command: |
+            apk update
+            apk add --update groff less py-pip
+            pip install awscli
+            docker build --build-arg VERSION_BUCKET=deb --build-arg FALCO_VERSION=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco-driver-loader:${CIRCLE_TAG}" docker/driver-loader
```
Comment (leodido) [inline .circleci/config.yml:603]:
```suggestion
            docker build --build-arg FALCO_IMAGE_TAG=${CIRCLE_TAG} -t "public.ecr.aws/falcosecurity/falco-driver-loader:${CIRCLE_TAG}" docker/driver-loader
```

**Review by leodido — COMMENTED** [review]
The `falco-driver-loader` image requires a `FALCO_IMAGE_TAG` build argument.

File: .circleci/config.yml [hunk 7]
```diff
@@ -471,6 +483,16 @@ jobs:
...
             aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
             docker push "public.ecr.aws/falcosecurity/falco:master"
+      - run:
+          name: Build and publish driver-loader to AWS
+          command: |
+            apk update
+            apk add --update groff less py-pip
+            pip install awscli
+            FALCO_VERSION=$(cat /build/release/userspace/falco/config_falco.h | grep 'FALCO_VERSION ' | cut -d' ' -f3 | sed -e 's/^"//' -e 's/"$//')
+            docker build --build-arg VERSION_BUCKET=deb-dev --build-arg FALCO_VERSION=${FALCO_VERSION} -t "public.ecr.aws/falcosecurity/falco-driver-loader:master" docker/driver-loader
```
Comment (leodido) [inline .circleci/config.yml:493]:
```suggestion
            docker build --build-arg FALCO_IMAGE_TAG=master -t "public.ecr.aws/falcosecurity/falco-driver-loader:master" docker/driver-loader
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -471,6 +483,16 @@ jobs:
             docker build --build-arg VERSION_BUCKET=deb-dev --build-arg FALCO_VERSION=${FALCO_VERSION} -t "public.ecr.aws/falcosecurity/falco:master" docker/falco
             aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
             docker push "public.ecr.aws/falcosecurity/falco:master"
+      - run:
+          name: Build and publish driver-loader to AWS
```
Comment (leodido) [inline .circleci/config.yml:487]:
```suggestion
          name: Build and publish driver-loader (dev) to AWS
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -461,6 +461,18 @@ jobs:
...
+            apk add --update groff less py-pip
+            pip install awscli
+            FALCO_VERSION=$(cat /build/release/userspace/falco/config_falco.h | grep 'FALCO_VERSION ' | cut -d' ' -f3 | sed -e 's/^"//' -e 's/"$//')
+            docker build --build-arg VERSION_BUCKET=bin-dev --build-arg FALCO_VERSION=${FALCO_VERSION} -t "public.ecr.aws/falcosecurity/falco-no-driver:master" docker/no-driver
+            docker tag public.ecr.aws/falcosecurity/falco-no-driver:master public.ecr.aws/falcosecurity/falco:master-slim
+            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/falcosecurity
+            docker push "public.ecr.aws/falcosecurity/falco-no-driver:master"
+            docker push "public.ecr.aws/falcosecurity/falco:master-slim"
       - run:
           name: Build and publish falco to AWS
```
Comment (leodido) [inline .circleci/config.yml:477]:
```suggestion
          name: Build and publish falco (dev) to AWS
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -461,6 +461,18 @@ jobs:
           at: /
       - checkout
       - setup_remote_docker
+      - run:
+          name: Build and publish no-driver to AWS
```
Comment (leodido) [inline .circleci/config.yml:465]:
```suggestion
          name: Build and publish no-driver (dev) to AWS
```

**Review by leodido — APPROVED** [review]
LGTM now :)

Thanks for approaching this Jonah, your help is important to us!

**Comment by jonahjon** [comment]
> LGTM now :)
> 
> Thanks for approaching this Jonah, your help is important to us!

Of course, actually getting this all fixed up finally here too

**Review by leogr — APPROVED** [review]
LGTM, 

Thank you, @jonahjon

**Review by leodido — APPROVED** [review]

**Pull request merged**